### ISSUE TITLE
Bad property type when null and date

### ISSUE THREAD

**Issue by sagikazarmark** [jolicode/jane#26]
When a property type is string or null and the format is date then:

- phpdoc is `@var \DateTime|mixed`
- there is no typehint

**Comment by sagikazarmark** [comment]
It turns out this is caused by the multiple type. I think the specific case of DateTime AND null should be handled correctly.

**Comment by sagikazarmark** [comment]
Although if #27 does not get fixed, this will still be an issue to be handled.

**Comment by joelwurtz** [comment]
Can you provide a schema example for this one ?

**Comment by sagikazarmark** [comment]
``` json
                "displayed": {
                    "type": ["string", "null"],
                    "format": "date-time"
                },
```

**Comment by sagikazarmark** [comment]
Another issue related to this:

Since the type detection is bad (`mixed`) in the normalizer the correct value will always be overwritten:

``` php

        if (is_object($object->getDisplayed())) {
            $value_8 = $object->getDisplayed()->format("Y-m-d\TH:i:sP");
        }
        if (!is_null($object->getDisplayed())) {
            $value_8 = $object->getDisplayed();
        }
```

**Comment by sagikazarmark** [comment]
This is probably a more serious issue than it seems: mixed type should never be applied if there is ANY identified type, but should be extracted one by one.

**Comment by joelwurtz** [comment]
Fixed in #31 and #30
